### PR TITLE
draft of example of BST auditory data

### PR DESCRIPTION
**Pull request by agramfort**
the idea is to replicate as much as possible:

http://neuroimage.usc.edu/brainstorm/Tutorials/Auditory

as least in sensor space to get clean evoked data as in:

http://neuroimage.usc.edu/brainstorm/Tutorials/Auditory?action=AttachFile&do=get&target=average_sensor.gif

so far I did not manage.

Hints:
- make use of their annotated bad segments. How to handle this in MNE workflow?
- investigate filtering issue cc @Eric89GXL

### CODE REVIEW

**Comment by agramfort** [comment]
anyone please take over. It's unlikely I'll have time to make progress here...

**Comment by Eric89GXL** [comment]
@agramfort who packaged the data for us? I don't see e.g. `events_bad_01.mat` in our archive.

File: examples/datasets/plot_brainstorm_auditory_data.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+raw = Raw(raw_fnames, preload=True)
+
+# set EOG channel
+raw.set_channel_types({'HEOG': 'eog', 'VEOG': 'eog', 'ECG': 'ecg'})
+
+# compute EOG and ECG projectors and add them to the data
+projs_eog = mne.preprocessing.compute_proj_eog(raw, n_grad=1, n_mag=1,
+                                               n_eeg=0)[0]
+raw.add_proj(projs_eog)
+projs_ecg = mne.preprocessing.compute_proj_eog(raw, n_grad=1, n_mag=1,
```
Comment (agramfort) [inline examples/datasets/plot_brainstorm_auditory_data.py]: compute_proj_eog -> compute_proj_ecg

**Comment by agramfort** [comment]
@mainakjas did it

you can update it and put it on aws if you want.

**Comment by Eric89GXL** [comment]
@agramfort first draft of management of bad segments added via `epochs.drop_bad_segments`. It didn't get rid of the contamination in the Standard case around 300 ms, though. Not sure what other preprocessing would do it. Maybe there is some CTF compensation that we're not applying?

**Comment by jasmainak** [comment]
@Eric89GXL : I'll do a review tomorrow. Hold on before you merge

**Comment by Eric89GXL** [comment]
This isn't near mergable yet, don't worry :)

**Comment by agramfort** [comment]
>
> @agramfort <https://github.com/agramfort> first draft of management of
> bad segments added via epochs.drop_bad_segments. It didn't get rid of the
> contamination in the Standard case around 300 ms, though. Not sure what
> other preprocessing would do it. Maybe there is some CTF compensation that
> we're not applying?
>
I have no idea but that puzzles me...

File: examples/datasets/plot_brainstorm_auditory_data.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+              % run_num for run_num in range(1, 3)]
+erm_fname = data_path + '/MEG/bst_auditory/S01_Noise_20131218_01_raw.fif'
+
+raw = Raw(raw_fnames, preload=True, add_eeg_ref=False)
+raw_erm = Raw(erm_fname, preload=True, add_eeg_ref=False)
+
+# set EOG channel
+raw.set_channel_types({'HEOG': 'eog', 'VEOG': 'eog', 'ECG': 'ecg'})
+
+# parse Brainstorm bad event data
```
Comment (jasmainak) [inline examples/datasets/plot_brainstorm_auditory_data.py]: I presume this needs to be part of a reader eventually? We can't have it in an example ...

Comment (agramfort) [inline examples/datasets/plot_brainstorm_auditory_data.py]: to me the handling of bad segments should be done at the raw level and the
epochs should be aware of it. It's much cleaner.

the bad segment should also be visible from the raw viewer.

**Comment by Eric89GXL** [comment]
@jasmainak you up for taking this one over now?

**Comment by jasmainak** [comment]
yep, I can take it over :)

**Comment by Eric89GXL** [comment]
Superceded by #2821 as far as I can tell

**Pull request closed without merging**